src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -178,24 +180,33 @@ private static Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibil
 
             if (type is INamedTypeSymbol namedType && namedType.OriginalDefinition != null)
             {
-                var originalDefinition = namedType.OriginalDefinition;
-
-                var ienumerableOfT = namedType.GetAllInterfacesIncludingThis().FirstOrDefault(
-                    t => t.OriginalDefinition.SpecialType == SpecialType.System_Collections_Generic_IEnumerable_T);
-
-                if (ienumerableOfT != null)
+                // if namedType contains a valid GetEnumerator method, we want collectionType to be the type of
+                // the "Current" property of this enumerator. For example:
+                // if namedType is a Span<Person>, collectionType should be Person.
+                var collectionType = namedType.GetMembers()
+                    .OfType<IMethodSymbol>()
+                    .FirstOrDefault(m => m.IsValidGetEnumerator() || m.IsValidGetAsyncEnumerator())
+                    ?.ReturnType?.GetMembers(WellKnownMemberNames.CurrentPropertyName)
+                    .OfType<IPropertySymbol>().FirstOrDefault(p => p.GetMethod != null)?.Type;
+
+                // This can happen for an un-implemented IEnumerable.
+                collectionType ??= namedType.GetAllInterfacesIncludingThis().FirstOrDefault(
+                        t => t.OriginalDefinition.SpecialType == SpecialType.System_Collections_Generic_IEnumerable_T)?.TypeArguments[0];
```

Comment:
In the higher up code you look for GEtEnumerator and GetAsyncEnumerator.  But here, you only chec for IEnumerable and not IAsyncEnumerable.  Intentional?

Reply:
also, would it not be sufficient to just do this check and skip the higher up check?

Reply:
> In the higher up code you look for GEtEnumerator and GetAsyncEnumerator. But here, you only chec for IEnumerable and not IAsyncEnumerable. Intentional?

Nope, I'd add a check for IAsyncEnumerable too.

Reply:
> also, would it not be sufficient to just do this check and skip the higher up check?

Both handles different cases. The main part of this is the first check, this second one won't be the common scenario.